client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/TcpTransportFactory.java

```diff
@@ -291,6 +291,12 @@ public Transport getTransport(Object key, Set<SocketAddress> failedServers, byte
 
    @Override
    public void releaseTransport(Transport transport) {
+      if (transport.isBusy()) {
```

Comment:
Why do you need this specifically? Is the pool trying to release transport instances still in use?